native/exe/ServiceControlLaunch/ServiceControlLaunch.cpp

```diff
@@ -105,9 +109,11 @@ void popupExceptionError(JNIEnv* env, jthrowable thrower, char* errTitle)
 	jstring throwStr = (jstring)env->CallObjectMethod(thrower, toStr);
 	const char* cThrowStr = env->GetStringUTFChars(throwStr, 0);
 	char *errStr = (char*) malloc(env->GetStringLength(throwStr) + 64);
-	sprintf(errStr, "An exception occured in Java:\n%s", cThrowStr);
+	size_t errStrlen = strlen(cThrowStr) + 64;  // Need to know this size for the sprintf_s
+	sprintf_s(errStr, errStrlen, "An exception occured in Java:\n%s", cThrowStr);
 	errorMsg(errStr, errTitle);
 	free(errStr);
+	env->ReleaseStringUTFChars(throwStr, cThrowStr);

```

Comment:
Nice catch, this was a memory leak...although nothing that would ever accumulate. :)

Reply:
Yeah,  I think we'd have much bigger problems before this caused an issue :)